bug: Background workspace diagnostics passes empty string as code, producing no diagn

## Description

Background workspace diagnostics passes empty string as code, producing no diagnostics for unopened files

## Required Labels

Priority: P0 | Type: bug

## Problem

bridge.analyze('', ['parse', 'diagnostics'], item.uri) passes an empty string as the source code. The Pike subprocess reads this code literally — it receives zero bytes to compile. Since workspace diagnostics targets unopened files (not in DocumentCache), this means background diagnostics silently produces zero results for every file it processes. The bug is invisible because the result is "no errors" rather than a crash.
- **expectedBehavior**: Background diagnostics should read the file content from disk via fs.readFile(item.uri) and pass that content to bridge.analyze(), producing real diagnostics for workspace files.

## Expected Behavior

Background diagnostics should read the file content from disk via fs.readFile(item.uri) and pass that content to bridge.analyze(), producing real diagnostics for workspace files.
- **suggestedApproach**: In processBatch(), read the file from disk before calling bridge.analyze(). Use the same pattern as pike-introspection.ts's readDocumentText() — fall back to uriToFsPath() + fs.readFile(). The batch loop should accumulate read errors but continue processing remaining files.

## Suggested Approach

In processBatch(), read the file from disk before calling bridge.analyze(). Use the same pattern as pike-introspection.ts's readDocumentText() — fall back to uriToFsPath() + fs.readFile(). The batch loop should accumulate read errors but continue processing remaining files.

## Affected Files

packages/pike-lsp-server/src/services/workspace-diagnostics.ts:206

## Acceptance

- [ ] Code change implemented
- [ ] Tests added/updated
- [ ] TypeScript compiles: bun run typecheck
- [ ] Smoke tests pass: scripts/test-agent.sh --fast
- [ ] No regression in existing tests

**Priority:** P0 | **Type:** bug | Auto-discovered by DGA Architect

**Compile failed:**

shell: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json 2>&1
  stdout: packages/pike-lsp-server/src/core/index.ts(9,50): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/index.ts(10,33): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/index.ts(11,34): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(8,33): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(166,20): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(206,21): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(212,26): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/features/advanced/code-actions.ts(25,24): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/inde

--- a/packages/pike-lsp-server/src/scenarios/workspace-idle-diagnostics.test.ts
+++ b/packages/pike-lsp-server/src/scenarios/workspace-idle-diagnostics.test.ts
@@ -7,14 +7,50 @@
 
 import { describe, it } from 'bun:test';
 import assert from 'node:assert/strict';
+import { mkdtempSync, writeFileSync, rmSync } from 'node:fs';
+import { tmpdir } from 'node:os';
+import { join } from 'node:path';
 import { WorkspaceDiagnosticsManager } from '../services/workspace-diagnostics.js';
 import { RequestScheduler } from '../services/request-scheduler.js';
-import { WorkspaceIndex } from '../workspace-index.js';
+import type { BridgeManager } from '../services/bridge-manager.js';
+import type { WorkspaceIndex } from '../workspace-index.js';
+
+/** Minimal mock bridge that records analyze() calls. */
+function createRecordingBridge() {
+  const analyzeCalls: Array<{ code: string; ops: string[]; filename?: string }> = [];
+
+  return {
+    analyze(code: string, ops: string[], filename?: string) {
+      if (filename !== undefined) {
+        analyzeCalls.push({ code, ops, filename });
+      } else {
+        analyzeCalls.push({ code, ops });
+      }
+      return Promise.resolve({});
+    },
+    isRunning() {
+      return true;
+    },
+    analyzeCalls,
+  };
+}
+
+/** Minimal BridgeManager mock. */
+function createMockBridgeManager(bridge: ReturnType<typeof createRecordingBridge>): BridgeManager {
+  return { bridge } as unknown as BridgeManager;
+}
+
+/** Create a mock WorkspaceIndex that returns given URIs. */
+function createMockWorkspaceIndex(uris: string[]): WorkspaceIndex {
+  return {
+    getAllDocumentUris: () => uris,
+  } as unknown as WorkspaceIndex;
+}
 
 describe('Scenario: workspace idle diagnostics', () => {
   it('should initialize with default options', () => {
     const scheduler = new RequestScheduler();
-    const workspaceIndex = new WorkspaceIndex();
+    const workspaceIndex = createMockWorkspaceIndex([]);
     const bridgeManager = null;
 
     const manager = new WorkspaceDiagnosticsManager({
@@ -33,7 +69,7 @@ describe('Scenario: workspace idle diagnostics', () => {
 
   it('should track idle state and yield to user activity', () => {
     const scheduler = new RequestScheduler();
-    const workspaceIndex = new WorkspaceIndex();
+    const workspaceIndex = createMockWorkspaceIndex([]);
 
     const manager = new WorkspaceDiagnosticsManager({
       scheduler,
@@ -52,7 +88,7 @@ describe('Scenario: workspace idle diagnostics', () => {
 
   it('should reset when indexing completes', () => {
     const scheduler = new RequestScheduler();
-    const workspaceIndex = new WorkspaceIndex();
+    const workspaceIndex = createMockWorkspaceIndex([]);
 
     const manager = new WorkspaceDiagnosticsManager({
       scheduler,
@@ -67,5 +103,82 @@ describe('Scenario: workspace idle diagnostics', () => {
     assert.strictEqual(stats.queueDepth, 0);
 
     manager.dispose();
+  });
+
+  it('should read file content from disk and pass to analyze (Issue #1319)', async () => {
+    const tempDir = mkdtempSync(join(tmpdir(), 'pike-ws-diag-'));
+    const filePath = join(tempDir, 'test.pike');
+    const fileContent = 'int main() { return 0; }\n';
+
+    try {
+      writeFileSync(filePath, fileContent);
+
+      const bridge = createRecordingBridge();
+      const bridgeManager = createMockBridgeManager(bridge);
+      const scheduler = new RequestScheduler();
+      const workspaceIndex = createMockWorkspaceIndex([filePath]);
+
+      const manager = new WorkspaceDiagnosticsManager({
+        scheduler,
+        workspaceIndex,
+        bridgeManager,
+        idleDelayMs: 10,
+        batchSize: 5,
+      });
+
+      // Trigger idle processing directly
+      manager.onIndexingComplete();
+
+      // Wait for idle processing to complete
+      await new Promise(resolve => setTimeout(resolve, 200));
+
+      // Verify analyze was called with actual file content, not empty string
+      assert.ok(bridge.analyzeCalls.length > 0, 'Expected analyze() to be called at least once');
+
+      const firstCall = bridge.analyzeCalls[0]!;
+      assert.strictEqual(
+        firstCall.code,
+        fileContent,
+        'analyze() should receive file content read from disk, not empty string'
+      );
+      assert.deepStrictEqual(firstCall.ops, ['parse', 'diagnostics']);
+    } finally {
+      rmSync(tempDir, { recursive: true, force: true });
+    }
+  });
+
+  it('should continue processing when a file read fails (Issue #1319)', async () => {
+    const tempDir = mkdtempSync(join(tmpdir(), 'pike-ws-diag-'));
+    const goodFilePath = join(tempDir, 'good.pike');
+    const badFilePath = join(tempDir, 'missing.pike');
+    const goodContent = 'int x = 1;\n';
+
+    try {
+      writeFileSync(goodFilePath, goodContent);
+      // Do NOT create badFilePath — simulates a missing file
+
+      const bridge = createRecordingBridge();
+      const bridgeManager = createMockBridgeManager(bridge);
+      const scheduler = new RequestScheduler();
+      const workspaceIndex = createMockWorkspaceIndex([goodFilePath, badFilePath]);
+
+      const manager = new WorkspaceDiagnosticsManager({
+        scheduler,
+        workspaceIndex,
+        bridgeManager,
+        idleDelayMs: 10,
+        batchSize: 5,
+      });
+
+      manager.onIndexingComplete();
+      await new Promise(resolve => setTimeout(resolve, 200));
+
+      // The good file should have been analyzed despite the bad file failing
+      const goodCall = bridge.analyzeCalls.find(c => c.filename === goodFilePath);
+      assert.ok(goodCall, 'Good file should be analyzed despite missing file in batch');
+      assert.strictEqual(goodCall!.code, goodContent);
+    } finally {
+      rmSync(tempDir, { recursive: true, force: true });
+    }
   });
 });

--- a/packages/pike-lsp-server/src/services/workspace-diagnostics.ts
+++ b/packages/pike-lsp-server/src/services/workspace-diagnostics.ts
@@ -9,6 +9,7 @@ import { Logger } from '@pike-lsp/core';
 import type { RequestScheduler } from './request-scheduler.js';
 import type { WorkspaceIndex } from '../workspace-index.js';
 import type { BridgeManager } from './bridge-manager.js';
+import { uriToFsPath } from '../utils/uri-path.js';
 
 const log = new Logger('WorkspaceDiagnostics');
 
@@ -199,11 +200,15 @@ export class WorkspaceDiagnosticsManager {
         requestClass: 'background',
         key: `workspace-diagnostics:${uriList.join(',')}`,
         run: async () => {
-          // Process each file in the batch
+          const fs = await import('node:fs/promises');
+
           for (const item of batch) {
             try {
-              // Quick parse-only analysis for background diagnostics
-              await bridge.analyze('', ['parse', 'diagnostics'], item.uri);
+              // Read file content from disk (unopened files aren't in DocumentCache)
+              const fsPath = uriToFsPath(item.uri);
+              const text = await fs.readFile(fsPath, 'utf-8');
+
+              await bridge.analyze(text, ['parse', 'diagnostics'], item.uri);
             } catch (err) {
               log.debug('Background diagnostic failed', {
                 uri: item.uri,

